packages/kolme/src/core/types/error.rs

```diff
@@ -19,6 +25,227 @@ pub enum KolmeError {
+    UnexpectedBlockHeight {
+        received: BlockHeight,
+        expected: BlockHeight,
+    },
+
+    #[error("Tried to add block with height {height}, but it's already present in the store")]
+    BlockAlreadyExists { height: BlockHeight },
+
+    #[error("Received block signed by processor {actual_processor}, but the real processor is {expected_processor}")]
+    InvalidBlockProcessor {
+        expected_processor: Box<PublicKey>,
+        actual_processor: Box<PublicKey>,
+    },
+
+    #[error("Unable to migrate contract for chain {chain:?}: contract isn't deployed")]
+    ContractNotDeployed { chain: ExternalChain },
+
+    #[error("Already have a deployed contract on {chain:?}")]
+    ContractAlreadyDeployed { chain: ExternalChain },
+
+    #[cfg(feature = "pass_through")]
+    #[error("No wait for pass-through contract is expected")]
+    UnexpectedPassThroughContract,
+
+    #[error("Persistent task exited unexpectedly")]
+    PersistentTaskExited,
+
+    #[error("Task exited with an error: {error}")]
+    TaskErrored { error: String },
+
+    #[error("Task panicked: {details}")]
+    TaskPanicked { details: String },
+
+    #[error("Trying to configure a Cosmos contract as a Solana bridge")]
+    CosmosBridgeConfiguredAsSolana,
+
+    #[error("Trying to configure a Solana program as a Cosmos bridge")]
+    SolanaBridgeConfiguredAsCosmos,
+
+    #[cfg(feature = "pass_through")]
+    #[error("Multiple pass-through bridges are not supported")]
+    MultiplePassThroughBridgesUnsupported,
+
+    #[cfg(feature = "pass_through")]
+    #[error("Pass-through bridge can't require Cosmos or Solana bridge contract")]
+    InvalidPassThroughBridgeType,
+
+    #[error("Expected exactly one message in the first block, but found a different number")]
+    InvalidGenesisMessageCount,
+
+    #[error("Invalid messages in first block")]
+    InvalidFirstBlockMessageType,
+
+    #[error("Listener panicked: {details}")]
+    ListenerPanicked { details: String },
+
+    #[error("Block parent mismatch: actual {actual}, expected {expected}")]
+    BlockParentMismatch {
+        actual: Box<BlockHash>,
+        expected: Box<BlockHash>,
+    },
+
+    #[error("Action ID mismatch: expected {expected}, found {found}")]
+    ActionIdMismatch {
+        expected: BridgeActionId,
+        found: BridgeActionId,
+    },
+
+    #[error("Validator {signer} already approved proposal {proposal_id}")]
+    AlreadyApprovedProposal {
+        signer: PublicKey,
+        proposal_id: AdminProposalId,
+    },
+
+    #[error("Conflicting block in DB at height {height} with hash {hash}")]
+    ConflictingBlockInDb { height: u64, hash: Sha256Hash },
+
+    #[error("Failed to execute signed Cosmos bridge transaction: {details}")]
+    CosmosExecutionFailed { details: String },
+
+    #[error("Timed out proposing and awaiting transaction {txhash}: {details}")]
+    TimeoutProposingTx { txhash: TxHash, details: String },
```

Comment:
It looks like `details` is just a stringified Elapsed value.  The original error message didn't include it, so I think it could be removed.